# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1523046639 25200
#      Fri Apr 06 13:30:39 2018 -0700
# Branch stable
# Node ID ad6dce96cbca9a9c7e886971b50db557d063616d
# Parent  d61f42a392b71faa391fcadc0fd87d1d377d5394
pkg.m: fix undefined variable error in install (bug #53567)

* pkg.m: Make sure tmp_dir is defined before the unwind_protect block.
Fixes error introduced by cset 0cd7076a97c8.

diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -391,16 +391,17 @@ function [local_packages, global_package
       endif
 
     case "install"
       if (isempty (files))
         error ("pkg: install action requires at least one filename");
       endif
 
       local_files = {};
+      tmp_dir = tempname ();
       unwind_protect
 
         if (octave_forge)
           [urls, local_files] = cellfun ("get_forge_download", files,
                                          "uniformoutput", false);
           [files, succ] = cellfun ("urlwrite", urls, local_files,
                                    "uniformoutput", false);
           succ = [succ{:}];
@@ -409,17 +410,17 @@ function [local_packages, global_package
             error ("pkg: could not download file %s from url %s",
                    local_files{i}, urls{i});
           endif
         else
           ## If files do not exist, maybe they are not local files.
           ## Try to download them.
           external_files_mask = ! cellfun (@exist, files, {"file"});
           if (any (external_files_mask))
-            [success, msg] = mkdir (tmp_dir = tempname ());
+            [success, msg] = mkdir (tmp_dir);
             if (success != 1)
               error ("pkg: failed to create temporary directory: %s", msg);
             endif
 
             for file_idx = find (external_files_mask)
 
               [~, fname, fext] = fileparts (files{file_idx});
               local_files{end+1} = fullfile (tmp_dir, [fname fext]);
